.github/workflows/ci.yml

```diff
@@ -327,6 +325,24 @@ jobs:
           python make.py python-ci-install
         timeout-minutes: 20
 
+      # We only save the libparsec lib when:
+      # - We are not in a github queue branch (they're a one time use so caching won't help)
+      # - We haven't already cached it.
+      - name: Save cached libparsec to be reused on later call
+        if: >-
+          steps.python-changes.outputs.run == 'true'
+          && steps.cache-libparsec.outputs.cache-hit != 'true'
+          && contains(github.ref, 'gh-readonly-queue') != 'true'
+        uses: actions/cache/save@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # pin v3.3.1
+        with:
+          # Key cache contains a hash of all the files that are used to produce _parsec.so
+          # Hence if we have a cache hit we know that there is no need for a rebuild !
+          key: ${{ matrix.os }}-${{ hashFiles('src/**', 'oxidation/libparsec/**', 'rust-toolchain.toml', 'Cargo.toml', 'Cargo.lock') }}-libparsec-python
```

Comment:
can we create a step dedicated to generate this key ?

This has two advantages:
- avoid producing an invalid value if one of the input file gets modified (or something is added in the directory) during the build
- (bonus) save a bit of time

Reply:
done